### PR TITLE
Added equals for tsType and made toString mandatory

### PR DESCRIPTION
**Pull request by ryan10132**
My use case here is to make it easier to test the library that depends on this one.

By the way, I created a typescript service generating library that is based of yours! Check it out and let me know what you think!

https://github.com/palantir/typescript-service-generator/

### CODE REVIEW

**Comment by vojtechhabarta** [comment]
I didn't have a chance to look at typescript-service-generator yet but in general I am happy that typescript-generator is used directly or in other tools.

**Comment by vojtechhabarta** [comment]
Please add these tests and make them work:
``` java
assertEquals(new TsType.StructuralType("Foo"), new TsType.StructuralType(new String("Foo")));
assertFalse(new TsType.StructuralType("Foo").equals(null));
```

Also please add `hashCode` method consistent with `equals`.

**Comment by ryan10132** [comment]
sure thing, done

File: typescript-generator-core/src/main/java/cz/habarta/typescript/generator/TsType.java [hunk 1]
```diff
@@ -16,10 +16,23 @@
     public static final AliasType DateAsNumber = new AliasType("DateAsNumber", "type DateAsNumber = number;");
     public static final AliasType DateAsString = new AliasType("DateAsString", "type DateAsString = string;");
 
+    @Override
+    public boolean equals(Object rhs) {
+        return rhs != null && this.getClass() == rhs.getClass() && this.toString() == rhs.toString();
```
Comment (vojtechhabarta) [inline typescript-generator-core/src/main/java/cz/habarta/typescript/generator/TsType.java]: You should not use `==` for string comparisons. That was the first test line I suggested to add.

**Pull request merged**